### PR TITLE
Khp3 6347 add an shr summary pannel

### PR DESCRIPTION
**Pull request by Omoshlawi**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
- Added OTP Authorization before pulling SHR Records 
- Moved SHR summary components from billing esm to shr esm

## Screenshots

https://github.com/user-attachments/assets/7346e71f-cce5-4e4e-8cd1-a93b5dc5fdc9

image

image

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-shr-app/src/hooks/visits.json [hunk 1]
```diff
@@ -0,0 +1,203 @@
+{
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/hooks/visits.json:1]: Should we have this in a mock some where

Comment (Omoshlawi) [reply packages/esm-shr-app/src/hooks/visits.json:1]: Like where? @donaldkibet

File: packages/esm-shr-app/src/index.ts [hunk 2]
```diff
@@ -19,11 +28,9 @@ export const importTranslation = require.context('../translations', false, /.jso
 
 export const shrPatientSummary = getSyncLifecycle(shrPatientSummaryComponent, options);
 
-// t('sharedhealthrecords', 'Shared Health Records')
-///export const shrSummaryDashboardLink = getSyncLifecycle(createDashboardLink({ ...dashboardMeta, moduleName }), options);
-
 export function startupApp() {
   registerBreadcrumbs([]);
+  registerFeatureFlag('shr-summary', 'SHR Summary', 'Adds authorization to pull a ptient SHR information');
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/index.ts:33]: Lets have one single source of truth for SHR work could we use this flag https://github.com/palladiumkenya/kenyaemr-esm-3.x/blob/9b5082f205c923ee07c5b48af2ea10de7f3f7cb4/packages/esm-billing-app/src/index.ts

File: packages/esm-shr-app/src/index.ts [hunk 3]
```diff
@@ -35,6 +42,10 @@ export const ReferralsDashboardLink = getSyncLifecycle(
   options,
 );
 
+export const shrSummaryDashboardLink = getFeatureFlag('shr-summary')
+  ? getSyncLifecycle(createDashboardLink({ ...shrSummaryDashboardMeta, moduleName }), options)
+  : undefined;
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/index.ts:47]: This isn't correct
```suggestion
export const shrSummaryDashboardLink = getSyncLifecycle(createDashboardLink({ ...shrSummaryDashboardMeta, moduleName }), options)
```

for this under routes.json we can have to toggle the feature flag
```json
{
      "name": "patient-chart-shr-summary-dashboard-link",
      "component": "shrSummaryDashboardLink",
      "slot": "patient-chart-dashboard-slot",
      "meta": {
        "columns": 1,
        "columnSpan": 1,
        "slot": "patient-chart-shr-summary-dashboard-slot",
        "path": "SHR Portal"
      },
      "featureFlag": "healthInformationExchange"
    },
```

File: packages/esm-shr-app/src/routes.json [hunk 4]
```diff
@@ -11,24 +11,26 @@
...
         "columnSpan": 1,
-        "slot": "patient-chart-shr-dashboard-slot",
-        "title": "SHARED HEALTH RECORDS",
-        "path": "SHR panel"
+        "slot": "patient-chart-shr-summary-dashboard-slot",
+        "path": "SHR Portal"
+      }
+    },
+    {
+      "name": "SHR Portal",
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/routes.json:29]: This is the place to have the feature flag for shr work.

File: packages/esm-shr-app/src/shr-summary/tables/shr-summary-table.component.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+} from '@openmrs/esm-patient-common-lib';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import usePatient from '../../hooks/usePatient';
+import SharedHealthRecordsSummary from '../../shrpatient-summary/shrpatient-summary.component';
+import styles from './shr-tables.scss';
+
+interface PatientSHRSummartTableProps {}
+
+const PatientSHRSummartTable: React.FC<PatientSHRSummartTableProps> = () => {
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/shr-summary/tables/shr-summary-table.component.tsx:19]:
```suggestion
const PatientSHRSummaryTable: React.FC<PatientSHRSummartTableProps> = () => {
```

Comment (donaldkibet) [reply packages/esm-shr-app/src/shr-summary/tables/shr-summary-table.component.tsx:19]: There is also typo on the props

**Review by donaldkibet — CHANGES_REQUESTED** [review]
Thanks @Omoshlawi 💪🏽

**Comment by donaldkibet** [comment]
ping @Omoshlawi please resolve the conflict. Would like to conclude work on this today. Thanks!

**Comment by Omoshlawi** [comment]
@donaldkibet  Done

File: packages/esm-shr-app/src/hooks/usePatientIdentifiers.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { FetchResponse, openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+import { PatientIdentifier } from '../types';
+
+const usePatientIdentifiers = (uuid: string) => {
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/hooks/usePatientIdentifiers.ts:5]: using `usePatient` hook from `@openmrs/esm-framework` that returns a `fhir.Patient` object we should be able to retrieve the identifiers, there is no need for this additional request

File: packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      return true;
+    },
+    {
+      message: 'Required',
+      path: ['receiver'],
+    },
+  );
+
+export function generateOTP(length = 5) {
+  var string = '0123456789';
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts:31]:
```suggestion
  const string = '0123456789';
```

Comment (donaldkibet) [reply packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts:31]: also good idea to have a more descriptive variable name here

File: packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  let OTP = '';
+  var len = string.length;
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts:33]: Its recommended we don't use `var` for declaring variables. Please use either `let` or `const`

File: packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts [hunk 9]
```diff
@@ -0,0 +1,96 @@
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import dayjs from 'dayjs';
+import { z } from 'zod';
+const PHONE_NUMBER_REGEX = /^(\+?254|0)((7|1)\d{8})$/;
```
Comment (donaldkibet) [inline packages/esm-shr-app/src/shr-summary/shr-summary.resource.ts:4]: good to have these in a separate file named `constants`

**Review by donaldkibet — APPROVED** [review]

**Comment by donaldkibet** [comment]
Merging this, please address the comments in a follow up PR.

**Pull request merged**